Rdmp.Dicom/Cache/Pipeline/Ordering/HierarchyBasedOrder.cs

```diff
@@ -84,10 +94,9 @@ public void InitialisePickers()
         internal void Retry(Item item)
         {
             if (parent is null)
-            {
-                if (retried.Contains(item))
```

Comment:
Are there any issues with this? I think retrying once and retrying many times should follow the same flow of logic?